src/api/core.rs

```diff
@@ -202,6 +202,30 @@ fn format_value_array_as_string(v: &[Value]) -> String {
     )
 }
 
+/// `Metadata` uses by `KeyValue` pairs to represent related metadata.
+#[cfg_attr(feature = "serialize", derive(Deserialize, Serialize))]
+#[derive(Clone, Debug, PartialOrd, PartialEq, Default)]
+pub struct Metadata(String);
```

Comment:
As this is only used in baggage, maybe it would add clarity to name this `BagggageMetadata` and move it to the baggage module?

Reply:
Yeah I think that make sense. Will make the changes.